### PR TITLE
Update bazel to 6.5.0

### PR DESCRIPTION
**Pull request by pzread**
Update bazel to 6.5.0, the latest maintenance version https://bazel.build/release and revert #16600

The LLVM is also currently using 6.1.5, so we are on the same major version.

### CODE REVIEW

File: build_tools/docker/prod_digests.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-gcr.io/iree-oss/base-arm64@sha256:7e101ab7ba20f23b6fba778f18bf2392f616967c716867dd6295358abf506a7b
+gcr.io/iree-oss/base@sha256:dc314b4fe30fc1315742512891357bffed4d1b62ffcb46258b1e0761c737b446
+gcr.io/iree-oss/samples@sha256:0cb4a27f8fce2e6c9a425b93ecd97bf3c5c3a8ec56929ab7ef005f51d4b1e054
+gcr.io/iree-oss/riscv@sha256:62e87bad3405d691ddba6f9be0ef44eeb60461a467c8d86f0842c81a1f97da79
+gcr.io/iree-oss/nvidia@sha256:82fa00b5cdda1b35634796cd0f88cb5d6d22d80328b94bfb51e5f2820598ba23
+gcr.io/iree-oss/emscripten@sha256:2dd4c52f1bb499ab365aad0111fe5538b685d88af38636b409b0cf6a576ab214
+gcr.io/iree-oss/android@sha256:66b92a1c920588a73d03316f26025407ea754bab93e8f9bfe40dbf6ed5fe6c7e
+gcr.io/iree-oss/base-bleeding-edge@sha256:c5f28883e6c570c20128fb37d7af3a00a25df3ce4e2b3a24c3a8dcd183182a27
+gcr.io/iree-oss/nvidia-bleeding-edge@sha256:81b3b5485f962c978bb7e5b2a6ded44ae4ef432048cafffe2b74fcf6dbe1bbca
+gcr.io/iree-oss/base-arm64@sha256:9daa1cdbbf12da8527319ece76a64d06219e04ecb99a4cff6e6364235ddf6c59
```
Comment (ScottTodd) [inline build_tools/docker/prod_digests.txt:9]: For a future refactoring / cleanup, we may want to pull Bazel out of the base image or fold more of these images together. If we only have one CI build job using it, I don't think it makes sense to have all of these other images and workflows dependent on that.

**Review by ScottTodd — APPROVED** [review]
LGTM if the builds pass. Thanks!

**Comment by pzread** [comment]
> LGTM if the builds pass. Thanks!

Thanks, also do you know why the extra test suite failed? https://github.com/openxla/iree/actions/runs/8086418421/job/22096503710

It shouldn't be touched by this change?

**Comment by ScottTodd** [comment]
> > LGTM if the builds pass. Thanks!
> 
> Thanks, also do you know why the extra test suite failed? https://github.com/openxla/iree/actions/runs/8086418421/job/22096503710
> 
> It shouldn't be touched by this change?

Yeah was just looking at that. I just added those new tests. Looks flaky 🤔 
That check should be non-blocking so feel free to ignore it.

**Pull request merged**